polygeist/tools/cgeist/Test/Verification/redstore.c

```diff
@@ -1,8 +1,11 @@
-// RUN: cgeist -O2 %s --function=* -S | FileCheck %s
+// RUN: clang++ -fsycl -fsycl-device-only -O2 -w -emit-mlir %s -o - | FileCheck %s
 
-extern int print(double);
+// FIXME: Recognize __restrict__ attribute without SYCL. 
```

Comment:
Do I understand correctly that we currently need SYCL compilation to understand `__restrict__`? 

Do you plan to fix that in a different PR before merging this one?

Reply:
> Do I understand correctly that we currently need SYCL compilation to understand __restrict__?

Yes, you understand it correctly, we only add attributes to SYCL functions. One reason is to avoid fixing all existing non SYCL test cases. 

> Do you plan to fix that in a different PR before merging this one?

To avoid fixing all existing non SYCL test cases, actually I don't plan to fix that, I can look into adding an option to allow adding attributes for non SYCL functions.

Reply:
Eventually we will need to process host code in MLIR so this limitation will need to be addressed but for now we can side step the problem.

Reply:
added option control.